src/main/kotlin/io/perfometer/data/DataSource.kt

```diff
@@ -0,0 +1,61 @@
+package io.perfometer.data
+
+import io.perfometer.data.csv.CsvParser
+import java.io.File
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.createType
+
+interface DataSource<T> {
+    val size: Int
+    operator fun get(index: Int): T
+}
+
+class ListDataSource<T>(private val list: List<T>) : DataSource<T> {
+    override val size: Int = list.size
+    override fun get(index: Int): T = list[index]
+}
+
+class CsvDataSource<T : Any>(
+    private val clazz: KClass<T>,
+    csvFilePath: String,
```

Comment:
I'd suggest having a convention on the constructor parameters order e.g.:
- first, parameters not assigned as property
- second, public `var/val` properties
- third, private `var/val` properties
Or the other way around, just to be consistent.

Reply:
done